lib/KOptionalText.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+
+  <span v-if="!textEmpty" :style="textStyle">
+    <template v-if="text">
+      {{ text }}
+    </template>
+    <template v-else>
+      <slot></slot>
+    </template>
+  </span>
+
+  <KEmptyPlaceholder v-else />
+
+</template>
+
+
+<script>
+
+  import KEmptyPlaceholder from './KEmptyPlaceholder.vue';
+
+  export default {
+    name: 'KOptionalText',
+    components: {
+      KEmptyPlaceholder,
+    },
+    props: {
+      /**
+       * Text to display
+       */
+      text: {
+        type: String,
+        default: null,
+      },
+      /**
+       * If provided, sets the styles of the text
+       */
+      textStyle: {
+        type: Object,
+        default: null,
+      },
+    },
+    computed: {
+      textEmpty() {
+        if (this.text && this.text.trim()) {
+          return false;
+        }
+
+        const slotsText = this.getSlotsText();
+        return !slotsText.trim();
+      },
+    },
+    methods: {
+      getSlotsText() {
+        if (!this.$slots.default || !this.$slots.default.length) {
+          return '';
+        }
+
+        return this.$slots.default
```

Comment:
I think it would be better to extract this code into an independent method and instead return the new method so that it is more clearer what the code is doing. At the moment the `getSlotsText`. method is doing one thing too many.